server/text_generation_server/models/custom_modeling/idefics2.py

```
@@ -679,6 +679,215 @@ def forward(self, image_hidden_states, attention_mask):
+                        real_images_inds
+                    ].contiguous()
+
+                patch_size = self.config.vision_config.patch_size
+                patches_subgrid = pixel_attention_mask.unfold(
+                    dimension=1, size=patch_size, step=patch_size
+                )
+                patches_subgrid = patches_subgrid.unfold(
+                    dimension=2, size=patch_size, step=patch_size
+                )
+                patch_attention_mask = (patches_subgrid.sum(dim=(-1, -2)) > 0).bool()
+
+                # Get sequence from the vision encoder
+                image_hidden_states = self.vision_model(
+                    pixel_values=pixel_values,
+                    patch_attention_mask=patch_attention_mask,
+                )
+
+                # Modality projection & resampling
+                image_hidden_states = self.connector(
+                    image_hidden_states,
+                    attention_mask=patch_attention_mask.view(pixel_values.size(0), -1),
+                )
+
+                all_states.append(image_hidden_states)
+            image_hidden_states = torch.stack(all_states, dim=0)
+            # TODO: remove when prefill image tokens are handled correctly
+            # * for now dummy tokens are added instead of the image tokens output byt the vision model
+            mask_size = (input_ids == self.config.image_token_id).sum().item()
+            unrolled_image_size = (
+                image_hidden_states.shape[1] * image_hidden_states.shape[2]
+            )
+            diff = mask_size - unrolled_image_size
+            if diff > 0:
+                print(
+                    f"Mask size {mask_size} is greater than the number of images {unrolled_image_size}."
+                )
+
+            if mask_size == unrolled_image_size:
+                inputs_embeds = self._merge_input_ids_with_image_features(
+                    input_ids, inputs_embeds, image_hidden_states
+                )
```

```suggestion
                inputs_embeds = self._merge_input_ids_with_image_features(
                    input_ids, inputs_embeds, image_hidden_states
                )
```

Let it crash if something is wrong here. We should NEVER do silent errors.

---

yes agreed! removed in the latest commits